modules/backup-azure/client.go

```diff
@@ -176,22 +172,27 @@ func (a *azureClient) PutObject(ctx context.Context, backupID, key string, data
 	return nil
 }
 
-func (a *azureClient) Initialize(ctx context.Context, backupID string) error {
+func (a *azureClient) Initialize(ctx context.Context, backupID, bucketName, bucketPath string) error {
 	key := "access-check"
 
-	if err := a.PutObject(ctx, backupID, key, []byte("")); err != nil {
+	if err := a.PutObject(ctx, backupID, key, bucketName, bucketPath, []byte("")); err != nil {
 		return errors.Wrap(err, "failed to access-check Azure backup module")
 	}
 
-	objectName := a.makeObjectName(backupID, key)
-	if _, err := a.client.DeleteBlob(ctx, a.config.Container, objectName, nil); err != nil {
-		return errors.Wrap(err, "failed to remove access-check Azure backup module")
+	containerName := a.config.Container
+	if bucketName != "" {
+		containerName = bucketName
+	}
+
+	objectName := a.makeObjectName(bucketPath, []string{backupID, key})
+	if _, err := a.client.DeleteBlob(ctx, containerName, objectName, nil); err != nil {
+		return errors.Wrap(err, "failed to remove access-check Azure backup module at"+objectName)
```

Comment:
we could use here a `errors.Wrapf` method to construct error

Reply:
why not? it improves readability and this is sprintf method is only invoked during errors:
```
return errors.Wrapf(err, "failed to remove access-check Azure backup module at %v", objectName)
```